delfin/task_manager/tasks/host_mapping.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import six
+from oslo_log import log
+
+from delfin import db
+from delfin.db.sqlalchemy.models import StorageHostGrpStorageHostRelation as \
+    StorageHostGrpRelation
+from delfin.db.sqlalchemy.models import VolumeGrpVolumeRelation, \
+    PortGrpPortRelation
+from delfin.i18n import _
+
+LOG = log.getLogger(__name__)
+
+
+class HostMappingTask(object):
+
+    def __init__(self):
+        pass
+
+    def _build_storgage_host_grp_relations(self, ctx, storage_id):
+        """ Builds storage host grp to host relations."""
+        storage_host_grps = db.storage_host_groups_get_all(
+            ctx, filters={"storage_id": storage_id})
+
+        storage_host_grp_relation_list = []
+        for storage_host_grp in storage_host_grps:
```

Comment:
Is it possible to optimize the way DS is built while storing? So that we can avoid o(n2) here

Reply:
As it is about group(having item list) to item relation building, I think 2 level traversals cant be avoided, any thought